dashboard/app/models/data_doc.rb

```diff
@@ -19,4 +19,29 @@ class DataDoc < ApplicationRecord
 
   validates_uniqueness_of :key, case_sensitive: false
   validate :validate_key_format
+
+  def file_path
+    Rails.root.join("config/data_docs/#{key.parameterize(preserve_case: false)}.json")
```

Comment:
super tiny nit: `preserve_case: false` looks like the default, so no need to specify it